### ISSUE TITLE
JavaScript syntax error with `v` flag in regexp

### ISSUE THREAD

**Issue by regseb** [mozilla/addons-linter#5462]
### Describe the problem and steps to reproduce it:

Add-ons Linter doesn't support the **`v`** flag ([`unicodeSets`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicodeSets)) in regular expression.

```javascript
const re = /foo/v;
```

### What happened?

> - Code: JS_SYNTAX_ERROR
> - Message: JavaScript syntax error (Parsing as module error: Invalid regular expression flag at line: 3 and column: 14) (Parsing as script error: Invalid regular expression flag at line: 3 and column: 14)
> - Description: There is a JavaScript syntax error in your code, which might be related to some experimental JavaScript features that aren't an official part of the language specification and therefore not supported yet. The validation cannot continue on this file.

### What did you expect to happen?

No error.

### Anything else we should know?

I think it's the JavaScript parser that doesn't support this new feature. If Add-ons Linter uses ESLint to parse JavaScript files, [ESLint _9.10.0_](https://eslint.org/blog/2024/09/eslint-v9.10.0-released/#other-notable-changes) supports `v` flag. Add-ons Linter uses ESLint _8.57.1_. Upgrading to v9 requires migrating to _FlatConfig_. The other eslint-plugin-no-unsanitized dependency has been compatible with ESLint v9 since [_4.1.0_](https://github.com/mozilla/eslint-plugin-no-unsanitized/releases/tag/4.1.0).
